engine/flink/process/src/main/scala/pl/touk/nussknacker/engine/process/FlinkBatchProcessRegistrar.scala

```diff
@@ -0,0 +1,205 @@
+
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.flink.api.common.functions._
+import org.apache.flink.api.java.RemoteEnvironment
+import org.apache.flink.api.scala.{ExecutionEnvironment, _}
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.runtime.execution.librarycache.FlinkUserCodeClassLoaders
+import org.apache.flink.streaming.api.windowing.time.Time
+import org.apache.flink.util.Collector
+import pl.touk.nussknacker.engine.Interpreter
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.exception.EspExceptionInfo
+import pl.touk.nussknacker.engine.api.test.TestRunId
+import pl.touk.nussknacker.engine.compiledgraph.part._
+import pl.touk.nussknacker.engine.flink.api.process.batch.{FlinkInputFormat, FlinkOutputFormat}
+import pl.touk.nussknacker.engine.flink.util.ContextInitializingFunction
+import pl.touk.nussknacker.engine.flink.util.metrics.InstantRateMeterWithCount
+import pl.touk.nussknacker.engine.graph.EspProcess
+import pl.touk.nussknacker.engine.process.FlinkBatchProcessRegistrar._
+import pl.touk.nussknacker.engine.process.compiler.{CompiledProcessWithDeps, FlinkProcessCompiler}
+import pl.touk.nussknacker.engine.process.util.{MetaDataExtractor, Serializers, UserClassLoader}
+import pl.touk.nussknacker.engine.splittedgraph.end.End
+import pl.touk.nussknacker.engine.splittedgraph.splittednode.SplittedNode
+import pl.touk.nussknacker.engine.util.metrics.RateMeter
+import pl.touk.nussknacker.engine.util.{SynchronousExecutionContext, ThreadUtils}
+
+import scala.concurrent.{Await, ExecutionContext}
+import scala.language.implicitConversions
+import scala.util.control.NonFatal
+
+class FlinkBatchProcessRegistrar(compileProcess: (EspProcess, ProcessVersion) => ClassLoader => CompiledProcessWithDeps,
+                                 enableObjectReuse: Boolean) extends LazyLogging {
+
+  implicit def millisToTime(duration: Long): Time = Time.of(duration, TimeUnit.MILLISECONDS)
+
+  def register(env: ExecutionEnvironment, process: EspProcess, processVersion: ProcessVersion, testRunId: Option[TestRunId] = None): Unit = {
+    Serializers.registerSerializers(env.getConfig)
+    if (enableObjectReuse) {
+      env.getConfig.enableObjectReuse()
+      logger.info("Object reuse enabled")
+    }
+
+    usingRightClassloader(env) {
+      register(env, compileProcess(process, processVersion), testRunId)
+    }
+  }
+
+  private def usingRightClassloader(env: ExecutionEnvironment)(action: => Unit): Unit = {
+    if (!env.getJavaEnv.isInstanceOf[RemoteEnvironment]) {
+      val flinkLoaderSimulation =  FlinkUserCodeClassLoaders.childFirst(Array.empty, Thread.currentThread().getContextClassLoader, Array.empty)
+      ThreadUtils.withThisAsContextClassLoader[Unit](flinkLoaderSimulation)(action)
+    } else {
+      action
+    }
+  }
```

Comment:
can we make it common with streaming?

Reply:
You're right. I should have extracted it.